spyder/plugins/onlinehelp/widgets.py

```diff
@@ -22,6 +23,189 @@
 from spyder.widgets.browser import WebBrowser
 
 
+# =============================================================================
+# Pydoc adjustments
+# =============================================================================
+# This is needed to prevent pydoc raise an ErrorDuringImport when
+# trying to import numpy.
+# See spyder-ide/spyder#10740
+DIRECT_PYDOC_IMPORT_MODULES = ['numpy', 'numpy.core']
+try:
+    from pydoc import safeimport
+
+    def spyder_safeimport(path, forceload=0, cache={}):
+        if path in DIRECT_PYDOC_IMPORT_MODULES:
+            forceload = 0
+        return safeimport(path, forceload=forceload, cache=cache)
+
+    pydoc.safeimport = spyder_safeimport
+except Exception:
+    pass
+
+
+# A reimplementation of pydoc._start_server to handle connection errors
+# for 'do_GET'.
+def _start_server(urlhandler, hostname, port):

```

Comment:
Let's move this function to another module in the same directory of this file called `pydoc_patch.py`.